experimental/packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -658,8 +660,10 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
       });
 
       const startTime = hrTime();
-      const metricAttributes: MetricAttributes =
-        utils.getOutgoingRequestMetricAttributes(attributes);
+      const metricAttributes: MetricAttributes = Object.assign(
+        utils.getOutgoingRequestMetricAttributes(attributes),
+        instrumentation._getConfig().customMetricAttributes?.()
```

Comment:
The custom hook has to be wrapped with `safeExecuteInTheMiddle` to prevent the exception from the custom hook being thrown. This is essential to avoid the malfunctioning instrumentation hook breaking the application. A good example would be how `applyCustomAttributesOnSpan` is called in this implementation.